app/models/reservation_validations.rb

```diff
@@ -45,6 +45,23 @@ def available
       "full time period requested.\n")
   end
 
+  def check_consecutive
+    return unless reserved? || requested?
+    return unless equipment_model
+    return unless equipment_model.max_checkout_length
+
+    # get all reservations of same user with same equipment model
+    # that are consecutive to this reservation
+    res = Reservation.for_reserver(reserver).for_eq_model(equipment_model)
+          .consecutive_with(self)
+
+    res.each do |r|
+      next unless duration + r.duration > equipment_model.max_checkout_length
```

Comment:
Does this mean you're only checking the sum of two reservations at a time? What if they had three consecutive reservations that together added up to greater than the max length but not any pair of them?